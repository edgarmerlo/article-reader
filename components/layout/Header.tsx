import React from 'react';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Container from '@mui/material/Container';

const ResponsiveAppBar = () => {
  return (
    <AppBar position="static">
      <Container maxWidth="xl">
        <Toolbar disableGutters>
          Article Reader v0.1.0
        </Toolbar>
      </Container>
    </AppBar>
  );
};
export default ResponsiveAppBar;
