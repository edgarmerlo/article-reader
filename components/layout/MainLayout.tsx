import React from 'react';
import Header from 'components/layout/Header';
import { ContentContainer } from 'components/layout/ContentContainer';

interface MainLayoutProps {
  children: React.ReactNode;
}

export const MainLayout: React.FunctionComponent<MainLayoutProps> = ({
  children,
}: MainLayoutProps) => {

  return (
    <>
      <Header></Header>
      <ContentContainer>
        {children}
      </ContentContainer>
    </>
  );
};
