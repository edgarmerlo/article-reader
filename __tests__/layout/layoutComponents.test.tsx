import { render } from '@testing-library/react';
import { ContentContainer } from 'components/layout/ContentContainer';
import Header from 'components/layout/Header';
import { MainLayout } from 'components/layout/MainLayout';

describe('Snapshots for layout components', () => {
  it('renders contentcontainer', () => {
    const { container } = render(<ContentContainer >Hi</ContentContainer>);
    expect(container).toMatchSnapshot();
  });

  it('renders header', () => {
    const { container } = render(<Header  />);
    expect(container).toMatchSnapshot();
  });

  it('renders article main layout', () => {
    const { container } = render(<MainLayout>Hi</MainLayout>);
    expect(container).toMatchSnapshot();
  });
});