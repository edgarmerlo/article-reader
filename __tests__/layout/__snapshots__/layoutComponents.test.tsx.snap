// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots for layout components renders article main layout 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
      >
        Article Reader v0.1.0
      </div>
    </div>
  </header>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-91mf4q-MuiPaper-root"
  >
    Hi
  </div>
</div>
`;

exports[`Snapshots for layout components renders contentcontainer 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-91mf4q-MuiPaper-root"
  >
    Hi
  </div>
</div>
`;

exports[`Snapshots for layout components renders header 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
      >
        Article Reader v0.1.0
      </div>
    </div>
  </header>
</div>
`;
